### PR TITLE
8: Add a contributing file

### PR DESCRIPTION
**Pull request by mosesnagar**
As commented on #8, I created that contributing file. 
That file need to be reviewed by the teammate to get comments about our working way.

Please feel free to left comments.

### CODE REVIEW

**Review by golan13 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## PRs and branches
+- On the PR's description, start with mentioned the issue. Use '#' and just choose the right issue or fill the issue's id.
+- The name of the new branch might start with #issueid#desciprtion_of_branch, so we will have one branch per issue. 
+For example: 153_creating_login_page
+
+
+## General Guidelines
+### linter and coding style
+We will follow a set of coding style rules and use linter to keep the order.
+Recommended flake8 or pylint (need to choose).
```
Comment (ifireball) [inline CONTRIBUTING.md:17]: Those are actually complementary

**Review by ifireball — COMMENTED** [review]
Generally LGTM - one small comment.

File: CONTRIBUTING.md (hunk 1 above)
Comment (mosesnagar) [reply CONTRIBUTING.md:17]: So do you recommend us to use both?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+### routes and controllers
+We need to keep routes short, in intent to follow rest-api rules.
+To keep the backend synchronized with the frontend, we need to talk one language.
+GET routes will always return templates(using render_template function) and POST routes always return a JSON.
+That way GET routes its for representing a view to the user, and POST routes its for getting data from the backend.
+
+
+### MVC design
```
Comment (ifireball) [inline CONTRIBUTING.md:27]: Given a comment I've made elsewhere about your team's ability to do work in parallel please reconsider concentrating things into files by technical subject like this.

With this design PRs that add models or routes will always conflict.

**Review by ifireball — APPROVED** [review]
Generally LGTM - I predict one of the things you wrote here may cause issues in the future but you can change that later once you've thought up a better solution.

**Review by Michal-Shamir — APPROVED** [review]

**Comment by ifireball** [comment]
Is this PR going to receive more updates or are we going to merge it now?

**Comment by mosesnagar** [comment]
> Is this PR going to receive more updates or are we going to merge it now?

I updated that PR, now it can be merged.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ That way GET routes its for representing a view to the user, and POST routes its
 
 
 ### MVC design
-Models - we will add new models into 'tutor/models.py' file.
-Views - we will add new views into 'tutor/templates' folder. We will use a layout as a base for all our templates. 
-Controllers - we will add controllers into 'tutor/controllers' folder. 
-Routing - we will add new routes into 'tutor/routing.py' with a route function named '%desciption%_route'.
+Models - we will add new models into 'tutor/models folder, one file for a model. We must do `vagrant halt` and then `vagrant up --provision` to keep the migrations synchronized.
```
Comment (ifireball) [inline CONTRIBUTING.md:28]: maybe you need something about migrations here?

Comment (mosesnagar) [reply CONTRIBUTING.md:28]: You are right, that is very important. Updated :)

**Pull request merged**